File: 2024/Day-05/Part-One.js
import { readFile } from 'node:fs/promises';

const fileContent = await readFile('input.txt', 'utf8');
let totalMiddlePagesNumber = 0;

let orderingRules = fileContent.trim().split('\n\n')[0];
let pages = fileContent.trim().split('\n\n')[1];

let rulesArray = orderingRules.split('\n')
let pageLine = pages.split('\n')
let correctLine = []

pageLine.forEach(line => {
    let lineWithoutComa = line.split(',')
    let str = ''
    let isCorrect = true

    for (let i = 0; i < lineWithoutComa.length; i++) {
        if( i + 1 < lineWithoutComa.length ) {
            str = lineWithoutComa[i + 1] + '|' + lineWithoutComa[i]
            if ( rulesArray.includes(str) )
                isCorrect = false
        }
    }
    if ( isCorrect ) correctLine.push(line)
});
correctLine.forEach(correctLine => {
    let array = correctLine.split(',')
    totalMiddlePagesNumber += getMiddleItemInArray(array)
})
function getMiddleItemInArray(array) {
    let middleIndex = parseInt(array.length / 2)
    
    return parseInt( array[ middleIndex ], 10 )
}

console.log(totalMiddlePagesNumber)

File: 2015/Day-03/Part-One.py
x = 0
y = 0

house_list = {(x, y)}
with open('input.txt') as file:
    for line in file:
        for el in line:
            if el == '>':
                x += 1
            elif el == '<':
                x -= 1
            elif el == '^':
                y += 1
            elif el == 'v':
                y -= 1
            else:
                continue
            
            house_list.add((x, y))

print(len(house_list))

